platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/internal/cc/impl/isolated/IsolatedProjectsBuildOperationsIntegrationTest.groovy

```diff
@@ -0,0 +1,377 @@
+            include("a")
+            include("b")
+        """
+
+        file("a/build.gradle") << """
+            plugins.apply(my.MyPlugin)
+        """
+
+        file("b/build.gradle") << """
+            // Not applying the plugin intentionally
+        """
+
+        initializeCache()
+
+        when: "subproject script is invalidated"
+        file("a/build.gradle") << """
+            println("project a updated")
+        """
+        fetchAllModels()
+
+        then: "emits sub project invalidation reason"
+        with(operations.only(ConfigurationCacheCheckFingerprintBuildOperationType).result) {
+            status == "PARTIAL"
+            buildInvalidationReasons == []
+            projectInvalidationReasons == [
+                [
+                    identityPath: ":a",
+                    invalidationReasons: [
+                        [message: "file 'a/build.gradle' has changed"]
+                    ]
+                ]
+            ]
+        }
+        outputContains("file 'a/build.gradle' has changed")
+    }
+
+    def "emits fingerprint check operation when invalidating multiple subprojects"() {
+        given:
+        withSomeToolingModelBuilderPluginInBuildSrc()
+
+        settingsFile """
+            include("a")
+            include("b")
+        """
+
+        file("a/build.gradle") << """
+            plugins.apply(my.MyPlugin)
+        """
+
+        file("b/build.gradle") << """
+            // Not applying the plugin intentionally
+        """
+
+        initializeCache()
+
+        when: "subproject scripts are invalidated"
+        file("a/build.gradle") << """
+            println("project a updated")
+        """
+        file("b/build.gradle") << """
+            println("project b updated")
+        """
+        fetchAllModels()
+
+        then: "emits subproject invalidation reason"
+        with(operations.only(ConfigurationCacheCheckFingerprintBuildOperationType).result) {
+            status == "PARTIAL"
+            buildInvalidationReasons == []
+            projectInvalidationReasons == [
+                [
+                    identityPath: ":a",
+                    invalidationReasons: [
+                        [message: "file 'a/build.gradle' has changed"]
+                    ]
+                ],
+                [
+                    identityPath: ":b",
+                    invalidationReasons: [
+                        [message: "file 'b/build.gradle' has changed"]
+                    ]
+                ]
+            ]
+        }
+        outputContains("file 'a/build.gradle' has changed")
+        outputDoesNotContain("file 'b/build.gradle' has changed")
+    }
+
+    def "emits fingerprint check operation when invalidating dependency"() {
+        given:
+        withSomeToolingModelBuilderPluginThatPerformsDependencyResolutionInBuildSrc()
+
+        settingsFile """
+            include("a")
+            include("b")
+        """
+
+        file("a/build.gradle") << """
```

Comment:
🌈  If you plan on doing any other passes on the PR, please consider replacing calls like things with their code-highlighting counterparts

```suggestion
        buildFile "a/build.gradle", """
```